### PR TITLE
Persistent lastOpenedPath

### LINKED ISSUE #309: Add a lastOpenedPath to blockdata.json
Keep track of the last opened path before close and store in `blockdata.json`.

### PR DESCRIPTION
**Pull request by firzanarmani**
Closes #309 

- Saves the lastOpenedPath in `BlockModel` JSON-parsed `blockdata.json`
- Change `storage` to store `BlockModel` instead of `BlockTree`
- Update tests
- Tweak `SideBarTreeView` to support this change

### CODE REVIEW

File: src/main/java/com/notably/MainApp.java [hunk 1]
```diff
@@ -86,22 +84,22 @@ public void init() throws Exception {
...
+            initialData = new BlockModelImpl();
         } catch (IOException e) {
             logger.warning("Problem while reading from the file. Will be starting with an empty BlockTree");
-            initialData = new BlockTreeImpl();
+            initialData = new BlockModelImpl();
         }
-        blockModel.setBlockTree(initialData);
+        blockModel.resetData(initialData);
         return new ModelManager(blockModel, suggestionModel, viewStateModel , userPrefs);
     }
```
Comment (kevinputera) [inline src/main/java/com/notably/MainApp.java]: Actually, @firzanarmani , would you consider making the `initModelManager` accept only `storage` and `userPrefs` as parameters? `blockModel`, `suggestionModel`, and `viewStateModel` can then be constructed inside the `initModelManager` function.

I feel like doing it this way makes the instantiation of `ModelManager` more straightforward. What do you think?

**Review by kevinputera — COMMENTED** [review]
LGTM. One suggestion: I think you haven't tested the scenario when the `lastOpenedPath` field does not relate to a path that exists in the block tree. Perhaps you'd want to consider that?

File: src/main/java/com/notably/MainApp.java (hunk 1 above)
Comment (firzanarmani) [reply src/main/java/com/notably/MainApp.java]: Sounds good to me!

**Review by kevinputera — APPROVED** [review]
LGTM!

**Pull request merged**